### DISCUSSION TITLE
Radio station adverts exposed as attribute

### DISCUSSION THREAD

**Discussion by mikejclark** [music-assistant/support#1935] [Feature requests and ideas]
Would be great to be able to mute a radio station when adverts are playing. I can see from the radio station within music assistant when an advert is playing (see attached image) but this is not currently available as an attribute 

![281200469-c06cb744-36a4-4ae0-8534-cd46225277d7](https://github.com/music-assistant/hass-music-assistant/assets/6953022/ae12dc0a-ca49-47e2-bf7f-c46cb00dbb23)

**Comment by OzGav** [comment]
Are you sure this isn’t shown in the stream_title attribute?

**Reply by mikejclark** [reply]
Unfortunately not. I don't see a stream_title attribute for my Google mini speaker

![Screenshot_20240222_150928_Home Assistant](https://github.com/music-assistant/hass-music-assistant/assets/6953022/ccc72b85-50e7-47dd-b6e8-37535803ad63)

**Reply by OzGav** [reply]
What version of the integration are you running?

**Reply by mikejclark** [reply]
I'm on latest of music assistant and home assistant

![Screenshot_20240224_100757_Home Assistant](https://github.com/music-assistant/hass-music-assistant/assets/6953022/50d24953-ab93-4ef8-bf87-b6f56d48d278)

**Reply by OzGav** [reply]
But what version of the HACS integration? It should be 2024.2.0

I just pulled up that radio station and it isn't broadcasting any track info

![image](https://github.com/music-assistant/hass-music-assistant/assets/19848947/e5280037-e387-422d-b957-f9253519da9e)

![image](https://github.com/music-assistant/hass-music-assistant/assets/19848947/5f332415-3008-4f01-b465-3b0bdfbabe87)

**Reply by mikejclark** [reply]
Just updated in hacs and I now see stream_title, but same as you it is null.
Something has changed in music assistant as I no longer see when an advert is playing

**Reply by OzGav** [reply]
Or the station no longer sends that info... Can you see it in some other software?

**Reply by mikejclark** [reply]
Yes, in the radio player android app, artist and track are displayed
![Screenshot_20240224_125745_Radioplayer](https://github.com/music-assistant/hass-music-assistant/assets/6953022/b6aed9d5-12fc-4149-a187-0cce004a119e)

**Reply by OzGav** [reply]
OK we will try and look at this at some point but it is low priority. Sorry.

**Reply by OzGav** [reply]
There is an attempt to identify adverts now. Have a look at the attributes when an ad plays. Obviously depends on the relevant station sending that info.

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
